### PR TITLE
Support other filesystems for Pinot Hadoop job

### PR DESCRIPTION
**Pull request by fx19880617**
Make Pinot Hadoop job to support other file systems like s3a.

### CODE REVIEW

File: pinot-hadoop/pom.xml [hunk 1]
```diff
@@ -43,13 +43,24 @@
...
             <artifactId>maven-shade-plugin</artifactId>
+            <version>3.2.1</version>
             <executions>
               <execution>
                 <phase>package</phase>
                 <goals>
                   <goal>shade</goal>
                 </goals>
                 <configuration>
+                  <relocations>
```
Comment (Jackie-Jiang) [inline pinot-hadoop/pom.xml]: Why shading these jars?

File: pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/BaseSegmentJob.java [hunk 2]
```diff
@@ -107,8 +107,14 @@ protected Path getPathFromProperty(String key) {
   protected List<Path> getDataFilePaths(Path pathPattern)
       throws IOException {
     List<Path> tarFilePaths = new ArrayList<>();
-    FileSystem fileSystem = FileSystem.get(_conf);
-    getDataFilePathsHelper(fileSystem, fileSystem.globStatus(pathPattern), tarFilePaths);
+    FileSystem fileSystem = FileSystem.get(pathPattern.toUri(), _conf);
+    _logger.info("Using filesystem: {}", fileSystem);
+    final FileStatus[] fileStatuses = fileSystem.globStatus(pathPattern);
```
Comment (Jackie-Jiang) [inline pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/BaseSegmentJob.java]: (nit) remove the final

File: pinot-hadoop/pom.xml (hunk 1 above)
Comment (fx19880617) [reply pinot-hadoop/pom.xml]: This is to relocate method call to use our shaded jar not from hadoop classpath. For Hadoop env, there could be older version of jackson and guava in classpath and cause NoSuchMethod exception.

Comment (Jackie-Jiang) [reply pinot-hadoop/pom.xml]: Can you please add some comments to the commit message?

File: pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentCreationJob.java [hunk 3]
```diff
@@ -108,15 +108,15 @@ protected boolean isDataFile(String fileName) {
       return true;
     }
     return fileName.endsWith(".avro") || fileName.endsWith(".csv") || fileName.endsWith(".json") || fileName
-        .endsWith(".thrift");
+        .endsWith(".thrift") || fileName.endsWith(".parquet") || fileName.endsWith(".orc");
```
Comment (jenniferdai) [inline pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentCreationJob.java]: You shouldn't need fileName.endsWith(".orc"), as the way ORC readers are currently configured go through JobConfigConstants.RECORD_READER_PATH

File: pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java [hunk 4]
```diff
@@ -148,7 +148,8 @@ public void setup(Context context)
     Preconditions.checkState(_localSegmentDir.mkdir());
     Preconditions.checkState(_localSegmentTarDir.mkdir());
 
-    _fileSystem = FileSystem.get(context.getConfiguration());
+    // Note: we made an assumption that input file and output path are using same filesystem.
+    _fileSystem = FileSystem.get(_hdfsSegmentTarDir.toUri(), context.getConfiguration());
```
Comment (jenniferdai) [inline pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java]: If I'm not misunderstanding, this one will always be local, right? It's assigned with FileOutputFormat.getWorkOutputPath(context)

Comment (fx19880617) [reply pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java]: This filesystem should be same as where we want to store the segment tars.
If we use s3a path as segment tar dir. Then this should be s3a filesystem.

File: pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java [hunk 5]
```diff
@@ -277,6 +277,12 @@ protected FileFormat getFileFormat(String fileName) {
     if (fileName.endsWith(".thrift")) {
       return FileFormat.THRIFT;
     }
+    if (fileName.endsWith(".parquet")) {
+      return FileFormat.PARQUET;
+    }
+    if (fileName.endsWith(".orc")) {
```
Comment (jenniferdai) [inline pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java]: Correct me if you are understanding differently / configuring it differently, but when I configure orc, I configure. Same with the comment I left yesterday in SegmentCreationJob.

record.reader.path: ${PATH_TO_ORCREADER} 
therefore, it skips the fileName.endsWith

    if (_recordReaderPath != null) {
      segmentGeneratorConfig.setRecordReaderPath(_recordReaderPath);
      segmentGeneratorConfig.setFormat(FileFormat.OTHER);
    } else {
      FileFormat fileFormat = getFileFormat(inputFileName);
      segmentGeneratorConfig.setFormat(fileFormat);
      segmentGeneratorConfig.setReaderConfig(getReaderConfig(fileFormat));
    }

Comment (fx19880617) [reply pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java]: yes, so basically it requires extra configs in order to read from orc/parquet files

Comment (jenniferdai) [reply pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java]: We count ORC as FileFormat.OTHER and set the record reader path to the ORCRecordReader, so you don't need to include the orc-specific logic for segmentcreation to work

**Pull request merged**